runtime/parachains/src/paras_inherent.rs

```diff
@@ -714,15 +738,11 @@ fn random_sel<X, F: Fn(&X) -> Weight>(
 
 	let mut weight_acc = 0 as Weight;
 
-	while !preferred_indices.is_empty() {
-		// randomly pick an index from the preferred set
-		let pick = rng.gen_range(0..preferred_indices.len());
-		// remove the index from the available set of preferred indices
-		let preferred_idx = preferred_indices.swap_remove(pick);
-
+	preferred_indices.shuffle(rng);
+	for preferred_idx in preferred_indices {
```

Comment:
Wouldn't it be quicker to use while loops here and below with `while weight_acc < weight_limit`? Oherwise we are forced to iterate over all indices, regardless of whether or not

Reply:
Actually, realizing this is intentional, since its possible the only item that can fit under the weight limit is the last item.